crates/rome_js_analyze/src/analyzers/nursery/no_nonoctal_decimal_escape.rs

```
@@ -0,0 +1,339 @@
+        _: &RuleContext<Self>,
+        RuleState {
+            diagnostics_text_range,
+            ..
+        }: &Self::State,
+    ) -> Option<RuleDiagnostic> {
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            diagnostics_text_range,
+            markup! {
+                r"Don't use `\8` and `\9` escape sequences in string literals."
+            },
+        ))
+    }
+
+    fn action(
+        ctx: &RuleContext<Self>,
+        RuleState {
+            kind,
+            replace_from,
+            replace_to,
+            replace_string_range,
+            ..
+        }: &Self::State,
+    ) -> Option<JsRuleAction> {
+        let mut mutation = ctx.root().begin();
+        let node = ctx.query();
+        let prev_token = node.as_js_string_literal_expression()?.value_token().ok()?;
+        let replaced = safe_replace_by_range(
+            prev_token.text().to_string(),
+            replace_string_range.to_owned(),
+            replace_to,
+        )
+        .ok()?;
+
+        let next_token =
+            JsSyntaxToken::new_detached(JsSyntaxKind::JS_STRING_LITERAL, &replaced, [], []);
+
+        mutation.replace_token(prev_token, next_token);
+
+        Some(JsRuleAction {
+            category: ActionCategory::QuickFix,
+            applicability: Applicability::MaybeIncorrect,
+            message: match kind {
+				FixSuggestionKind::Refactor => {
+					markup! ("Replace "<Emphasis>{replace_from}</Emphasis>" with "<Emphasis>{replace_to}</Emphasis>". This maintains the current functionality.").to_owned()
+				}
+				FixSuggestionKind::EscapeBackslash => {
+					markup! ("Replace "<Emphasis>{replace_from}</Emphasis>" with "<Emphasis>{replace_to}</Emphasis>" to include the actual backslash character." ).to_owned()
+				}
+			},
+            mutation,
+        })
+    }
+}
+
+fn safe_replace_by_range(
+    mut target: String,
+    range: std::ops::Range<usize>,
+    replace_with: &str,
+) -> Result<String, Box<dyn Error>> {
```

Considering the fact **we always suppress the error** (the code does `safe_replace_by_range().ok()?`), I suggest returning `Option` here and use `debug_assert` in the places where we do the checks.